src/test/java/com/example/DsStore/service/CustomerServiceTest.java

```diff
@@ -0,0 +1,160 @@
+package com.example.DsStore.service;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.example.DsStore.exceptions.IdNotFoundException;
+import com.example.DsStore.exceptions.ResourceNotFoundException;
+import com.example.DsStore.repositories.CustomerRepo;
+import com.example.DsStore.serviceImpl.CustomerServiceImpl;
+import com.example.DsStore.entities.Customer;
+
+@ExtendWith(MockitoExtension.class)
+public class CustomerServiceTest {
+
+	@Mock
+	CustomerRepo customerRepo;
+
+	@InjectMocks
+	CustomerServiceImpl customerService;
+
+	/**
+	 * Create Customer Test
+	 *
+	 */
+	@Test
+	public void testCreateCustomer() {
+		Customer customer = new Customer(1, "Amit", "Punjab", 12345);
+		when(customerRepo.save(any(Customer.class))).thenReturn(customer);
+		Customer newCustomer = customerService.createCustomer(customer);
+		assertNotNull(newCustomer);
+		assertEquals(customer, newCustomer);
+	}
+
+	/**
+	 * Update Product Test
+	 *
+	 * @throws IdNotFoundException id not found
+	 */
+	@Test
+	public void testUpdateProduct() throws IdNotFoundException {
+		Customer customer = new Customer(1, "Amit", "Punjab", 12345);
+		when(customerRepo.save(any(Customer.class))).thenReturn(customer);
+		when(customerRepo.findById(anyInt())).thenReturn(Optional.of(customer));
+
+		customer.setCustomerName("Pulkit");
+		Customer updatedCustomer = customerService.updateCustomer(customer, customer.getCustomerId());
+
+		assertNotNull(updatedCustomer);
+		assertEquals("Pulkit", customer.getCustomerName());
+
```

Comment:
Looks good, but could you check back and see if we can do `verify()` for verifying repository calls made by our service?
If we can/should... Let's do it for all service tests?

Reply:
yes we can, in some tests i have verified i will do it for all.